meta-mender-core/classes/mender-helpers.bbclass

```diff
@@ -350,19 +357,50 @@ def get_extra_parts_by_id(d, id):
 
 def get_extra_parts_labels(d):
     labels = []
-    parts = get_extra_parts(d)
-    if parts:
-        for part in parts:
-            s = part.find("--label")
-            if s > 0:
-                e = part.find("--", s + 2)
-                labels.append(part[s + len("--label="):e])
+    ids = get_extra_parts_flags(d)
+    if ids:
+        for id in ids:
+            labels.append(get_extra_parts_label(d, id))
     return ' '.join(labels)
 
+def get_extra_parts_label(d, id = None):
+    part = get_extra_parts_by_id(d, id)
+    if part:
+        s = part.find("--label")
+        if s > 0:
+            e = part.find("--", s + 2)
+            return part[s + len("--label="):e]
+    return ""
+
 def get_extra_parts_wks(d):
     final_parts = []
     parts = get_extra_parts(d) or {}
     if parts:
         for part in parts:
             final_parts.append("part --ondisk \"$ondisk_dev\" --align \"$alignment_kb\" {}".format(part))
     return '\n'.join(final_parts)
+
+def get_extra_parts_fstab_opts(d, id = None):
+    parts_fstab_flags = d.getVarFlags("MENDER_EXTRA_PARTS_FSTAB") or {}
+    extra_parts_flags = (d.getVar('MENDER_EXTRA_PARTS') or "").split()
+    if parts_fstab_flags:
+        if id in parts_fstab_flags:
+            return parts_fstab_flags[id]
+
+    return "auto default"
+
+def get_extra_parts_fstab(d):
+    out = []
+    extra_parts_offset = mender_get_extra_parts_offset(d)
+    device = d.getVar('MENDER_STORAGE_DEVICE_BASE')
+    for part in get_extra_parts_flags(d):
+       label = get_extra_parts_label(d, part)
+       fstype_opts = get_extra_parts_fstab_opts(d, part).split()
+       if len(fstype_opts) < 2:
+            bb.warn("MENDER_EXTRA_PARTS_FSTAB[%s] invalid." % part)
+            continue
```

Comment:
I think this should either be a failure (`bbfatal`) or a graceful fallback to using `auto default`. Personally I think I vote for the former. This could perhaps be a reason for moving that logic into `get_extra_parts_fstab_opts` instead of here.

What will happen now is that the build will continue, but you will be missing a whole partition. Definitely not what you want.

Reply:
@kacf ok thanks a lot for comment. I'll update in next few days :+1: